### ISSUE TITLE
Facebook preview doesn't work for /fun but works for /betacal

### ISSUE THREAD

**Issue by OneHwang** [Shift2Bikes/shiftcal#139]
Example - We see a 500 error for http://shift2bikes.org/fun/event-10549 
It works fine for http://shift2bikes.org/betacal/event-10549

**Comment by fool** [comment]
I'm seeing identical responses for both at present - can you confirm?  If there is a 500 error that sounds like something I can help debug.

**Comment by OneHwang** [comment]
Yes, I confirm. There is a problem on the Shift server - probably with one of the htaccess files - that prevents facebook preview from working correctly when the site is served from /fun.

**Comment by fool** [comment]
could you give me reproduction steps, please?  I thought you meant in the
browser.

**Comment by OneHwang** [comment]
1. Make sure you are signed in to facebook
2. In your timeline, try sharing http://shift2bikes.org/fun/event-10549
3. You will see there's a 500 error
4. Open http://shift2bikes.org/betacal/event-10549 in your browser
5. If you click on "share on facebook", you will see that the preview works fine
6. Open https://developers.facebook.com/tools/debug/ and enter http://shift2bikes.org/fun/event-10549
7. You will see that there's a 302 redirect to http://www.shift2bikes.org/erratum/500.html

**Comment by fool** [comment]
Thanks!  I am so looking forward to getting to throw away this legacy config that none of us understand completely!

TL;DR I haven't figured it out in 45 minutes of looking  and I'm about to ask CanvasDreams Support for some help.  

I think we'll need their help since:
-  we can't see the full apache error logs as far as I can tell, 
- and I think the config is split-brained between cpanel and config files files in the directory and am not sure I'm seeing all of it.

My presumption is that there is some difference in the config between betacal and cal and I am guessing this might be the cause of or related to the error.

Here's what I have found thus far:

You see that redirect because of this directive in /.htaccess: 

`ErrorDocument 500 http://www.shift2bikes.org/erratum/500.html`

...which I read as a genuine server error that we'll need to see the aforementioned apache error log to diagnose.  I can't find the *server* error log - just site errors.

This is a bit odd:

```
RewriteRule ^fun\/?$ "http\:\/\/www\.shift2bikes\.org\/fun\/www\/"
RewriteRule ^fun.\/?$ "http\:\/\/www\.shift2bikes\.org\/fun\/www\/"
```
- I don't think it's needed anymore, what do you think?  I don't see a "fun/www" directory.  There's a similar one for betacal, but it *does* have a `www` directory. 

Anyway - I tried commenting it out briefly and didn't observe any change in the facebook debugger.

There's also something I don't quite understand, which I assume y'all figured out and put in place in `/fun/.htaccess`.  Since it's the same for `fun/` and `betacal/` so I don't think it's the problem.

I'll let you know what canvasdreams' support can help us come up with.

**Comment by fool** [comment]
Update: Support wrote back and is trying to help us generate error logs - we are supposed to be able to see them but they are not being created

**Comment by OneHwang** [comment]
There is a "fun/www" directory - try `cd ~/fun/www` and you will see the contents. We have `RewriteRule ^fun\/?$ "http\:\/\/www\.shift2bikes\.org\/fun\/www\/"` because we don't want to give public access to the `app` folder or other files in the `fun/`.

**Comment by fool** [comment]
I can't see it on the filesystem, @OneHwang 👍 

```
# ls -1 fun/
./
../
addevent.js
calendarimages/
crawl.php
crawl.php.bak
css/
datepicker.js
delete_event.php
error_log.txt@
eventimages@
events.php
fonts/
helpers.js
.htaccess
ics.php
images/
index_170313.html
index.css
index.html
index-vue.html
js/
main.js
manage_event.php
retrieve_event.php
test.php
testrewrite/
vue-main.js
```

It's there in betacal/ - but not in fun/ : 

```
# cd public_html/fun/www
-jailshell: cd: public_html/fun/www: No such file or directory
shift2b@shift2bikes.org [~]# cd public_html/betacal/www
shift2b@shift2bikes.org [~/public_html/betacal/www]#
```

**Comment by sdobz** [comment]
I'm so glad that this is stumping others, I thought I was going crazy.

Remember that the public_html directories are links:
```
shift2b@shift2bikes.org [~]# ls -la public_html/fun
lrwxrwxrwx 1 shift2b shift2b 10 May 18  2017 public_html/fun -> ../fun/www/
```

The `~/public_html/fun` directory **IS** the `~/fun/www` directory.

**Comment by OneHwang** [comment]
I created a new directory, ~/fun2 that does what /fun is supposed to, and changed .htaccess rules so that /fun now redirects to /fun2. Facebook preview works.

**Comment by fool** [comment]
hmm, twasn't me; I haven't changed anything about redirects in recent
memory.  I tried to help you debug this issue months ago, but that's the
last time I looked at things and I changed nothing at the time.

Here's what's in cpanel at canvashost:


Everything else is on the filesystem in your directories (e.g. /fun) so I
don't want to change your files.

**Comment by fool** [comment]
if changes are made, could they please be documented here so we can
understand them?  I'm in crunch mode editing the print calendar right now
for the next 3 days, and if the admin tool breaks I'll be kinda fucked...

**Comment by OneHwang** [comment]
I did `git pull origin master` within /betacal and /fun directories. I did `cp -r betacal fun2`. The shift2bikes website then showed images as broken links. Then, for /betacal and /fun2, I changed the app/config.php file:

```
$IMAGEDIR = '/cal/eventimages';
$IMAGEPATH = '/cal/eventimages';
```
The shift2bikes website now displays the images properly. However, for reasons I don't understand, the facebook preview only shows text and not images.
